### ISSUE TITLE
Terminology: why --synchronous?

### ISSUE THREAD

**Issue by flx42** [hpc/mpifileutils#346]
Apologies for being pedantic today :)

Why is direct I/O behind a flag called `--synchronous`? Shouldn't it be `--direct`, or `--unbuffered`?
Using `O_DIRECT` for `open(2)` will not impact whether `read(2)` and `write(2)` are synchronous. 

Excluding sockets with O_NONBLOCK, synchronous / asynchronous is usually a property of the syscall itself. I believe the aio API requires direct (buffered) I/O, but that's not the case anymore with io_uring. 
You can have asynchronous direct I/O (io_uring, aio), asynchronous buffered I/O (io_uring), synchronous direct I/O (read/write with O_DIRECT), synchronous buffered I/O (read/write).

Also, all of these modes should not be confused with `O_SYNC`/`fdatasync`/`fsync`, which is mostly orthogonal to the above :).

**Comment by adammoody** [comment]
Yes, it's probably worth changing the option name here.  I don't recall whether we originally had other ideas on what this should do, but it's only meaning to turn on O_DIRECT at this point.  Switching to ```--direct``` would be more accurate.  We'll consider that change for the next major release in v0.11.

**Comment by adammoody** [comment]
Working to rename to ``--direct`` in https://github.com/hpc/mpifileutils/pull/380.  My plan is to keep ``--synchronous`` working for another release, but undocumented, and then print a note that the option name is changing.
